language/polkavm-move-native/src/types.rs

```diff
@@ -51,22 +51,45 @@ pub struct MoveSignerVector {
 ///
 /// The pointer must be to static memory and never mutated.
 #[repr(C)]
+#[derive(Clone, Copy)]
 pub struct MoveType {
     pub name: StaticTypeName,
     pub type_desc: TypeDesc,
     pub type_info: *const TypeInfo,
 }
 pub const MOVE_TYPE_DESC_SIZE: u64 = core::mem::size_of::<MoveType>() as u64;
 
+// Needed to make the MoveType, which contains raw pointers,
+// Sync, so that it can be stored in statics for test cases.
+unsafe impl Sync for MoveType {}
+
+impl core::fmt::Debug for MoveType {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("MoveType")
+            .field("type", &self.type_desc)
+            .finish()
+    }
+}
+
 /// # Safety
 ///
 /// The pointer must be to static memory and never mutated.
 #[repr(C)]
+#[derive(Clone, Copy, Debug)]
 pub struct StaticTypeName {
     pub ptr: *const u8,
     pub len: u64,
 }
 
+impl StaticTypeName {
+    pub unsafe fn as_ascii_str(&self) -> &str {
+        core::str::from_utf8_unchecked(core::slice::from_raw_parts(
+            self.ptr,
```

Comment:
this won't work, unless you copy also pointer value from guest into some owned vec and only then doing string conversion

Reply:
It's not used, I'll remove it.